startup_script/run.sh

```diff
@@ -104,7 +110,35 @@ function run_install_optional_components_script() {
   if ! is_version_at_least "2.3" || [[ -z "$USER_DATAPROC_COMPONENTS" ]]; then
     return
   fi
-  source "${BDUTIL_DIR}/install_optional_components.sh"
+
+  export BDUTIL_DIR="/usr/local/share/google/dataproc/bdutil"
+  cat > ./install_optional_components.sh <<EOF
```

Comment:
I still think this is silly.  subshells are intended for this purpose.

https://screenshot.googleplex.com/4wcnyohSd4jtLzU